File: server/test/e2e/page-objects/StatisticsPage.ts
import setLocation from "../setLocation";
import {browser, By, element} from "protractor";
import {TribeStatisticsStyles} from "./Styles";

const statisticsElement = element(By.className(TribeStatisticsStyles.className));

async function waitForPage() {
    browser.wait(() => statisticsElement.isPresent(), 2000);
}

export default {
    statisticsElement,
    async goTo(tribeId: string) {
        await setLocation(`/${tribeId}/statistics/`);
        await waitForPage();
    },
    waitForPage
}

File: e2e/src/endToEndTest/kotlin/com/zegreatrob/coupling/e2e/StatisticsPage.kt
package com.zegreatrob.coupling.e2e

import com.zegreatrob.coupling.model.tribe.TribeId
import com.zegreatrob.coupling.e2e.external.webdriverio.WebdriverBrowser

object StatisticsPage : StyleSyntax {
    override val styles = loadStyles("stats/TribeStatistics")
    private val teamStatisticsStyles = loadStyles("stats/TeamStatistics")
    private val pairReportTableStyles = loadStyles("stats/PairReportTable")

    val rotationNumber by teamStatisticsStyles.getting()
    val pairReport by pairReportTableStyles.getAll()

    suspend fun goTo(tribeId: TribeId) {
        WebdriverBrowser.setLocation("/${tribeId.value}/statistics")
        waitForPage()
    }

    suspend fun waitForPage() {
        element().waitToExist()
    }
}
